src/Compilers/CSharp/Test/Emit/CodeGen/SwitchTests.cs

```diff
@@ -3041,6 +3042,108 @@ .maxstack  3
             var pid = ((NamedTypeSymbol)comp.GlobalNamespace.GetMembers().Single(s => s.Name.StartsWith("<PrivateImplementationDetails>", StringComparison.Ordinal)));
             var member = pid.GetMembers(PrivateImplementationDetails.SynthesizedStringHashFunctionName).Single();
             Assert.Equal(Accessibility.Internal, member.DeclaredAccessibility);
+
+            compVerifier = CompileAndVerify(text, options: TestOptions.ReleaseExe.WithModuleName("MODULE"), expectedOutput: "True");
+
+            compVerifier.VerifyIL("Test.M", """
+{
+  // Code size      181 (0xb5)
+  .maxstack  3
+  .locals init (string V_0, //value
+                int V_1,
+                char V_2)
+  IL_0000:  ldstr      ""
+  IL_0005:  stloc.0
+  IL_0006:  ldarg.0
+  IL_0007:  brfalse.s  IL_0021
+  IL_0009:  ldarg.0
+  IL_000a:  ldstr      "C#"
+  IL_000f:  callvirt   "int string.IndexOf(string)"
+  IL_0014:  ldc.i4.m1
+  IL_0015:  beq.s      IL_0021
+  IL_0017:  ldarg.0
+  IL_0018:  ldc.i4.0
+  IL_0019:  ldc.i4.2
+  IL_001a:  callvirt   "string string.Remove(int, int)"
+  IL_001f:  starg.s    V_0
+  IL_0021:  ldarg.0
+  IL_0022:  brfalse.s  IL_0073
+  IL_0024:  ldarg.0
+  IL_0025:  brfalse    IL_00ad
+  IL_002a:  ldarg.0
+  IL_002b:  call       "int string.Length.get"
+  IL_0030:  stloc.1
+  IL_0031:  ldloc.1
+  IL_0032:  brfalse.s  IL_00ad
+  IL_0034:  ldloc.1
+  IL_0035:  ldc.i4.1
+  IL_0036:  bne.un.s   IL_00ad
+  IL_0038:  ldarg.0
+  IL_0039:  ldc.i4.0
+  IL_003a:  call       "char string.this[int].get"
```

Comment:
:memo: we only need to do a character check, no need to follow up with a string check, since we're dealing with strings of length==1.